tests/unit/classifier/AveragedPeceptron_unittest.cc

```diff
@@ -0,0 +1,86 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ */
+#include <gtest/gtest.h>
+
+#include <functional>
+#include <rxcpp/rx-lite.hpp>
+#include <shogun/lib/Signal.h>
+
+#include "environments/LinearTestEnvironment.h"
+#include <shogun/base/some.h>
+#include <shogun/classifier/AveragedPerceptron.h>
+#include <shogun/evaluation/ContingencyTableEvaluation.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/labels/BinaryLabels.h>
+
+using namespace shogun;
+
+extern LinearTestEnvironment* linear_test_env;
+
+TEST(AveragedPerceptron, continue_training_consistency)
+{
+	auto env = linear_test_env->getBinaryLabelData();
+	auto features = wrap(env->get_features_train());
+	auto labels = wrap(env->get_labels_train());
+	auto test_features = wrap(env->get_features_test());
+	auto test_labels = wrap(env->get_labels_test());
+
+	auto avrg_perceptron = some<CAveragedPerceptron>();
+	avrg_perceptron->set_labels(labels);
+	avrg_perceptron->train(features);
+
+    // reference for completly trained model
+	auto results = avrg_perceptron->apply(test_features);
+
+	index_t iter = 0;
+
+	// prematurely stopped at 1 iterations
+	auto avrg_perceptron_stop = some<CAveragedPerceptron>();
+
+	std::function<bool()> callback = [&iter]() {
```

Comment:
with lambdas in general you should use `auto` instead of a function pointer with `std::function`. Here it doesn't matter too much as it is a test but in the library code use `auto` for lambdas when you can.